packages/core/src/agent/__tests__/AgentMessage.test.ts

```diff
@@ -43,32 +43,51 @@ describe('AgentMessage', () => {
 
       const message = JsonTransformer.fromJSON(json, CustomProtocolMessage)
 
-      await expect(MessageValidator.validate(message)).resolves.toBeUndefined()
+      await expect(message).toBeInstanceOf(CustomProtocolMessage)
     })
 
-    it('successfully validates if the message type minor version is higher than the supported message type', async () => {
+    it('successfully validates if the message type minor version is higher than the supported message type', () => {
       const json = {
         '@id': 'd61c7e3d-d4af-469b-8d42-33fd14262e17',
         '@type': 'https://didcomm.org/fake-protocol/1.8/message',
       }
 
       const message = JsonTransformer.fromJSON(json, CustomProtocolMessage)
 
-      await expect(MessageValidator.validate(message)).resolves.toBeUndefined()
+      expect(message).toBeInstanceOf(CustomProtocolMessage)
     })
 
     it('throws a validation error if the message type major version differs from the supported message type', async () => {
-      expect.assertions(1)
-
       const json = {
         '@id': 'd61c7e3d-d4af-469b-8d42-33fd14262e17',
         '@type': 'https://didcomm.org/fake-protocol/2.0/message',
       }
 
-      const message = JsonTransformer.fromJSON(json, CustomProtocolMessage)
-
-      await expect(MessageValidator.validate(message)).rejects.toMatchObject([
+      expect(() => JsonTransformer.fromJSON(json, CustomProtocolMessage)).toThrowError(ClassValidationError)
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      let thrownError: any
+      try {
+        JsonTransformer.fromJSON(json, CustomProtocolMessage)
+      } catch (e) {
+        thrownError = e
+      }
```

Comment:
This will mainly help in other place where we are calling `expect` in a catch blocks. If the try block never throws the expects will never get called and the tests will succeed. We could also solve this using `expect.assertions(n)`.

Reply:
maybe we can make that a separate issue?

Reply:
and then also go through the code and see where this is applicable and make changes so we keep it consistent

Reply:
opened #874 for this yesterday